src/game/UINationSelectDlg.cpp

```diff
@@ -333,3 +216,104 @@ void CUINationSelectDlg::Render() {
         CN3Base::s_lpD3DDev->SetTextureStageState(0, D3DTSS_ALPHAARG2, dwTexAlpha2);
     }
 }
+
+bool CUINationSelectDlg::ReceiveMessage(CN3UIBase * pSender, DWORD dwMsg) {
+    if (!pSender) {
+        return false;
+    }
+
+    if (!m_bTransitionActive && dwMsg == UIMSG_BUTTON_CLICK) {
+        if (pSender == (CN3UIBase *)m_pBtnCloseKa || pSender == (CN3UIBase *)m_pBtnCloseEl) {
+            CGameProcedure::s_pSocket->Disconnect();
+            CGameProcedure::ProcActiveSet((CGameProcedure *)CGameProcedure::s_pProcLogIn);
+            CGameProcedure::s_pProcLogIn->m_bLogIn = true;
+            return true;
+        } else if (pSender == (CN3UIBase *)m_pBtnNextKa || pSender == (CN3UIBase *)m_pBtnNextEl) {
+            ChangeNation((pSender != (CN3UIBase *)m_pBtnNextKa));
+            ButtonsSetEnable(false);
+            return true;
+        } else if (pSender == (CN3UIBase *)m_pBtnSelectionKa || pSender == (CN3UIBase *)m_pBtnSelectionEl) {
+            m_eCurNation = (pSender == (CN3UIBase *)m_pBtnSelectionKa) ? NATION_KARUS : NATION_ELMORAD;
+            m_pProcNationSelectRef->MsgSendNationSelect(m_eCurNation);
+            return true;
+        }
+    }
+
+    return true;
+}
+
+void CUINationSelectDlg::InitResources() {
+    std::string szBuff;
+    if (m_pTextNoticeEl) {
+        // El Morad is a nation that was formed by King Manes...
+        ::_LoadStringFromResource(IDS_NATIONSELECT_DESCRIPTION_EL, szBuff);
+        m_pTextNoticeEl->SetString(szBuff);
+        m_pTextNoticeEl->SetFont(m_pTextNoticeEl->GetFontName(), m_pTextNoticeEl->GetFontHeight() - 2, FALSE, FALSE);
+    }
+    if (m_pTextNoticeKa) {
+        // Karus is a new nation that was established in Adonis...
+        ::_LoadStringFromResource(IDS_NATIONSELECT_DESCRIPTION_KA, szBuff);
+        m_pTextNoticeKa->SetString(szBuff);
+        m_pTextNoticeKa->SetFont(m_pTextNoticeKa->GetFontName(), m_pTextNoticeKa->GetFontHeight() - 2, FALSE, FALSE);
+    }
```

Comment:
Is it how it behaves officially with getting the font size and - 2? I haven't checked, but I believe if the font doesn't fit in the description box, maybe it's using the wrong font or text wrapping isn't working correctly.

In a situation we still need to decrease the font size, as much as it's not ideal, I suggest modifying the relevant game asset UIF file to reduce confusion when users set font size in the UIF files, but it doesn't reflect correctly in game.
```suggestion
    if (m_pTextNoticeEl) {
        // El Morad is a nation that was formed by King Manes...
        ::_LoadStringFromResource(IDS_NATIONSELECT_DESCRIPTION_EL, szBuff);
        m_pTextNoticeEl->SetString(szBuff);
    }
    if (m_pTextNoticeKa) {
        // Karus is a new nation that was established in Adonis...
        ::_LoadStringFromResource(IDS_NATIONSELECT_DESCRIPTION_KA, szBuff);
        m_pTextNoticeKa->SetString(szBuff);
```

Reply:
I checked the assembly and saw that they don't do that. I agree with your thoughts. Here is a PR that fixes the actual file: https://github.com/ko4life-net/ko-assets/pull/6

Reply:
Also see related commit: https://github.com/ko4life-net/ko/pull/125/commits/cd13eb401fcfefd301d4809ff7b1944999c91861

Reply:
Good change. Nice work! 🚀